### DISCUSSION TITLE
I would really like to see the variable  $CONFIG_PRODUKT_NAME on each page of the Freetz WebIF

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#903] [Unerfüllte Wünsche] [answered]
I would really like to see the variable  $CONFIG_PRODUKT_NAME on each page of the Freetz WebIF
Where I really miss it, is on the Firmware upgrade page.

I would then know which image I need to upload (7490, 7590, 7530AX).

Sometimes I need to upgrade 70 Fritzboxes and I jump immediately to the upgrade page /cgi-bin/update/firmware.cgi
I then don't know which firmware I need to use and have to jump forth to another page and back.

I tried to do it myself, but that didn't work.

**Comment by fraternl** [comment]
cat make/pkgs/mod/files/root/usr/share/skin/cuma/skin.sh
```
skin_head() {
        local title=$1 id=$2
        local hname="$(hostname -s|html)"
        [ "$hname" != "fritz" ] && hname="&#64;${hname}&nbsp;" || hname=""
        cat << EOF
<title>Freetz&nbsp;${hname}&ndash; $title</title>
<link rel="stylesheet" type="text/css" href="/style/cuma/base.css">
<link rel="stylesheet" type="text/css" href="/style/colorscheme.css">
EOF
        _cgi_print_extra_styles

        # The width of the whole cgi: There are 40px border (left+right)
        # and 225px for the menu area.
        let _cgi_total_width=_cgi_width+40+225

        # If there is no menu, we make the space available to the content.
        [ -z "$id" ] && let _cgi_width+=225
}
```
Instead of "Freetz" I'd like the content of CONFIG_PRODUKT_NAME 
I have no idea about the pitfalls in cgi

I would think that  this would work

```
.
.
local pname="${CONFIG_PRODUKT_NAME}"
.
<title>${pname}&nbsp;${hname}&ndash; $title</title>
.

```

I obviously do not understand how it works.

**EDIT**
After noticing the word "Freetz" was still in place, I noticed that the actual titlebar is displayed somewhere else in that file...
Now compiling and checking...

**Comment by fraternl** [comment]
diff make/pkgs/mod/files/root/usr/share/skin/cuma/skin.sh
```
26d25
<       local pname="$(html "_$CONFIG_PRODUKT_NAME")"
40c39
< <div class="titlebar"><a href="https://freetz-ng.github.io" target="_blank" class="logo">${pname}</a>&nbsp;${hname}<a href="/cgi-bin/about.cgi" target="_blank">&ndash;</a>&nbsp;<span class="title">$title</span>$help &nbsp; &nbsp; $([ "$isauth" = 1 ] && echo "<small>Session timeout: $(date -d @$(( $MOD_HTTPD_SESSIONTIMEOUT + $(date +%s) )) +'%H:%M:%S')</small>") </div>
---
> <div class="titlebar"><a href="https://freetz-ng.github.io" target="_blank" class="logo">Freetz</a>&nbsp;${hname}<a href="/cgi-bin/about.cgi" target="_blank">&ndash;</a>&nbsp;<span class="title">$title</span>$help &nbsp; &nbsp; $([ "$isauth" = 1 ] && echo "<small>Session timeout: $(date -d @$(( $MOD_HTTPD_SESSIONTIMEOUT + $(date +%s) )) +'%H:%M:%S')</small>") </div>

```

It seems the variable $CONFIG_PRODUKT_NAME is unavailable in that environment as the **underscore** I set **did** display.

![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/93ef546d-e001-4d18-96df-1087357f8470)

I will get it out of a file somehow...

**Comment by fraternl** [comment, marked as answer]
Well, this is working:

diff make/pkgs/mod/files/root/usr/share/skin/cuma/skin.sh

```
26d25
<       local pname="$(grep FREETZ_TYPE_PREFIX= /etc/.config | awk -F= '{print $2}' | tr '_' ' ' | tr -d '"'|html)"
40c39
< <div class="titlebar"><a href="https://freetz-ng.github.io" target="_blank" class="logo">${pname}</a>&nbsp;${hname}<a href="/cgi-bin/about.cgi" target="_blank">&ndash;</a>&nbsp;<span class="title">$title</span>$help &nbsp; &nbsp; $([ "$isauth" = 1 ] && echo "<small>Session timeout: $(date -d @$(( $MOD_HTTPD_SESSIONTIMEOUT + $(date +%s) )) +'%H:%M:%S')</small>") </div>
---
> <div class="titlebar"><a href="https://freetz-ng.github.io" target="_blank" class="logo">Freetz</a>&nbsp;${hname}<a href="/cgi-bin/about.cgi" target="_blank">&ndash;</a>&nbsp;<span class="title">$title</span>$help &nbsp; &nbsp; $([ "$isauth" = 1 ] && echo "<small>Session timeout: $(date -d @$(( $MOD_HTTPD_SESSIONTIMEOUT + $(date +%s) )) +'%H:%M:%S')</small>") </div>

```

![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/cf0756aa-a1d2-4afb-8239-5b2dc93e5261)

**Comment by fraternl** [comment]
Can this be done more elegantly, like first sourcing the variables?
It is now also dependent of the, _I believe optional_, creation of /etc/.config

```
local pname="$(grep FREETZ_TYPE_PREFIX= /etc/.config | awk -F= '{print $2}' | tr '_' ' ' | tr -d '"'|html)"
```

Or should I create a file like /etc/.product ??
It seems /etc/.version is created during compile time (and so is /etc/.config).

I also found the sudoku in **/usr/share/about.txt** hidden under the dash
**/usr/share/about.txt** is a compressed file

cat make/pkgs/mod/files/root/usr/mww/cgi-bin/about.cgi
```
#!/bin/sh
echo -en "Content-Type: text/html; charset=ISO-8859-1\r\n\r\n"
uudecode /usr/share/about.txt -o - | zcat
```

**Reply by MaxineMuster** [reply]
You can do the same like "Mod-Info" (.../cgi-bin/status/mod/info.cgi) and source
`. /etc/freetz_info.cfg` and then use `$FREETZ_INFO_BOXTYPE`

Or, like the "Status" page, use `$CONFIG_PRODUKT_NAME` after sourcing `. /bin/env.mod.rcconf`, e.g.
`local pname="$(. /bin/env.mod.rcconf  && echo $CONFIG_PRODUKT_NAME | html)"`


As a note: `$CONFIG_PRODUKT_NAME`will give the full name, not only the number (e.g. "FRITZ!Box 7590") but you might just add an "`| tr -dc [0-9]`" to get rid of everything but the numbers if you like it short.
